src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    public class MethodBody
+    {
+        protected MethodBody() { }
+
+        private byte[] _IL = null;
+        private ExceptionHandlingClause[] _exceptionHandlingClauses = null;
+        private LocalVariableInfo[] _localVariables = null;
+#if CORECLR
```

Comment:
I am wondering whether we should split all these related types to pure abstract shared base class, and internal RuntimeXXX class. Similar to how we have done RuntimeLocalVariableInfo.

Reply:
I also think that its really good idea. The code will be much clear and binding code(attached) will be in src
Let me know if that's the final decision ?

Reply:
Yes, I like it too.